pkg/common/k8s_utils.go

```diff
@@ -187,3 +187,19 @@ func FindDeploymentStatusCondition(conditions []appsv1.DeploymentCondition, cond
 
 	return nil
 }
+
+func HasLabel(obj metav1.Object, key string) bool {
+	labels := obj.GetLabels()
+	if labels == nil {
```

Comment:
nit: no need to check if it is `nil`. Reading from a `nil` map reference is safe. It will always return `false`